var/spack/repos/builtin/packages/nmap/package.py

```diff
@@ -32,25 +33,52 @@ class Nmap(AutotoolsPackage):
 
     variant("liblua", default=True, description="Enable lua (required by all of NSE)")
     variant("ncat", default=True, description="Enable ncat")
-    variant("ndiff", default=True, description="Enable ndiff")
     variant("nping", default=True, description="Enable nping")
-    variant("zenmap", default=True, description="Enable zenmap")
     variant("nmap-update", default=False, description="Enable nmap-update")
 
-    depends_on("openssl@1.1:", when="@7.50:")
+    # Release notes for 7.3: "Ensure Nmap builds with OpenSSL 3.0 using no
+    # deprecated API functions."
+    depends_on("openssl@3.0.7:", when="@7.93:")
+
+    # previous verisons use a few deprecated openssl apis
+    depends_on("openssl@1.1:", when="@7.50:7.92")
     depends_on("openssl@:1.0.9", when="@:7.49")
 
+    # nmap includes vendors a few libraries, but we should build against spack's
+    # versions where appropriate. see their build guide at
+    # https://nmap.org/book/inst-source.html#inst-configure. Also see:
+    # https://github.com/nmap/nmap/issues/1602
+    #
+    # Any that are included "for convinience" we should provide.
+    #
+    # The recursive builds to build the vendored dependencies is also broken on
+    # darwin (something races); using our own deps works around that issue.
+    #
+    # Specifically, something is touching libpcre/configure during the build,
+    # which causes the recursive `make`'s configure.status --recheck pass to
+    # attempt to rerun configure. Rerunning configure fails due and asks to run
```

Comment:
adding a comment for myself so I can commit these comment changes using the webui

```suggestion
    # attempt to rerun configure. Rerunning configure fails and asks you to run
```